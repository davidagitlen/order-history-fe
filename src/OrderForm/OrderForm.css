.order-form {
  display: flex;
  flex-direction: column;
  height: 450px;
  width: 300px;
  margin: 45px 0px 0px 0px;
}

input {
  font-size: 18px;
  height: 35px;
  margin: 10px 0px 10px 0px;
}

.order-form button {
  font-size: 18px;
  height: 35px;
  width: 300px;
  margin: 5px 0px 0px 0px;
  background-color: #5FDD70;
}

.order-form button:hover {
  font-style: italic;
  color: blue;
}