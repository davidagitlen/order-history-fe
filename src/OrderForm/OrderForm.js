import React, { Component } from 'react';
import './OrderForm.css'

class OrderForm extends Component {
  constructor() {
    super();
    this.state = {
      img : 'broken-link.png',
      name : '',
      description : '',
      price : '',
      id : Date.now()
    }
  }

  handleChange = e => {
    this.setState({[e.target.name] : e.target.value});
  }

  resetChanges = () => {
    this.setState({img : 'broken-link.png', name : '', description : '', price : '', id : Date.now()});
  }

  handleClick = e => {
    e.preventDefault();
    this.props.addNewPurchase(this.state);
    this.resetChanges();
  }


  render() {
    return (
      <form className='order-form'>
        <input 
          type='text'
          name='name'
          value={this.state.name}
          placeholder='Product name'
          onChange={this.handleChange}
          />
        <input
          type='text'
          name='description'
          value={this.state.description}
          placeholder='Product description'
          onChange={this.handleChange}
          />
        <input 
          type='number'
          name='price'
          value={this.state.price}
          placeholder='Produce price'
          onChange={this.handleChange}
          />
        <button onClick={this.handleClick}>Add Purchase</button>
      </form>
    )
  }
}

export default OrderForm;