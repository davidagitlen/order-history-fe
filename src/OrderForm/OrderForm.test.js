import React from 'react';
import { shallow } from 'enzyme';
import OrderForm from './OrderForm';

describe('OrderForm', () => {

  let wrapper, mockAddNewPurchase, mockRemovePurchase;

  beforeEach(() => {
    mockAddNewPurchase = jest.fn();
    mockRemovePurchase = jest.fn();
    wrapper = shallow(<OrderForm addNewPurchase={mockAddNewPurchase} removePurchase={mockRemovePurchase}/>);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on handleChange', () => {
    const mockEvent = {
      target : {
        name : 'name',
        value : 'Suspicious package'
      }
    }
    const expected = 'Suspicious package';

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should reset state when resetChanges is called', () => {
    global.Date.now = jest.spyOn(global.Date, 'now').mockImplementation(() => 100);
    const initialValues = { img: 'broken-link.png', name: 'Blib', description: 'a blob', price: 10000, id: 100 };
    const expected = {img : 'broken-link.png', name : '', description : '', price : '', id : 100};
    wrapper.setState(initialValues);
    expect(wrapper.state('name')).toEqual('Blib');
    wrapper.instance().resetChanges();
    expect(wrapper.state()).toEqual(expected);
  });

  it('should call resetChanges when handleClick is called', () => {
    wrapper.instance().resetChanges = jest.fn();
    const mockEvent = {preventDefault : jest.fn()};
    wrapper.instance().handleClick(mockEvent);
    expect(wrapper.instance().resetChanges).toHaveBeenCalled();
  });

})