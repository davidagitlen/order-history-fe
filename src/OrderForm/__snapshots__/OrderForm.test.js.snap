// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm should match the snapshot with all data passed in correctly 1`] = `
<form
  className="order-form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Product name"
    type="text"
    value=""
  />
  <input
    name="description"
    onChange={[Function]}
    placeholder="Product description"
    type="text"
    value=""
  />
  <input
    name="price"
    onChange={[Function]}
    placeholder="Produce price"
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Purchase
  </button>
</form>
`;
